diff --git a/pipenv/utils.py b/pipenv/utils.py
index 60c12cb..ef2555d 100644
--- a/pipenv/utils.py
+++ b/pipenv/utils.py
@@ -218,7 +218,7 @@ def prepare_pip_source_args(sources, pip_args=None):
 def actually_resolve_deps(
     deps, index_lookup, markers_lookup, project, sources, verbose, clear, pre, req_dir=None
 ):
-    from .vendor.packaging.markers import default_environment
+    from packaging.markers import default_environment
     from .patched.notpip._internal import basecommand
     from .patched.notpip._internal.cmdoptions import no_binary, only_binary
     from .patched.notpip._internal.req import parse_requirements
diff --git a/tests/integration/conftest.py b/tests/integration/conftest.py
index d71f0f8..dd31adf 100644
--- a/tests/integration/conftest.py
+++ b/tests/integration/conftest.py
@@ -1,14 +1,14 @@
 import json
 import os
 import warnings
+import requests
+import six
+import toml
 
 import pytest
 
 from pipenv._compat import TemporaryDirectory, Path
 from pipenv.vendor import delegator
-from pipenv.vendor import requests
-from pipenv.vendor import six
-from pipenv.vendor import toml
 
 if six.PY2:
     class ResourceWarning(Warning):
